File: stepik-java-api/src/main/java/org/stepik/api/queries/recommendations/StepikRecommendationsGetQuery.java
package org.stepik.api.queries.recommendations;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.objects.recommendations.Recommendations;
import org.stepik.api.queries.StepikAbstractGetQuery;
import org.stepik.api.urls.Urls;

/**
 * @author meanmail
 */
public class StepikRecommendationsGetQuery extends StepikAbstractGetQuery<StepikRecommendationsGetQuery, Recommendations> {
    public StepikRecommendationsGetQuery(StepikAbstractAction stepikAction) {
        super(stepikAction, Recommendations.class);
    }

    @NotNull
    public StepikRecommendationsGetQuery course(long course) {
        addParam("course", course);
        return this;
    }

    @NotNull
    @Override
    protected String getUrl() {
        return Urls.RECOMMENDATIONS;
    }

    @Override
    protected boolean isCacheEnabled() {
        return false;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/recommendations/StepikRecommendationsGetQuery.kt
package org.stepik.api.queries.recommendations

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.objects.recommendations.Recommendations
import org.stepik.api.queries.StepikAbstractGetQuery

class StepikRecommendationsGetQuery(stepikAction: StepikAbstractAction) :
        StepikAbstractGetQuery<StepikRecommendationsGetQuery, Recommendations>(stepikAction,
                Recommendations::class.java) {
    
    fun course(course: Long): StepikRecommendationsGetQuery {
        addParam("course", course)
        return this
    }
    
    override val url = "${stepikAction.stepikApiClient.host}/api/recommendations"
    
    override fun isCacheEnabled() = false
}
